quickstarts/microsoft.web/function-app-storage-private-endpoints/main.bicep

```diff
@@ -0,0 +1,442 @@
+          groupIds: [
+            'file'
+          ]
+        }
+      }
+    ]
+  }
+}
+
+resource storageTablePrivateEndpoint 'Microsoft.Network/privateEndpoints@2021-02-01' = {
+  name: privateEndpointStorageTableName
+  location: location
+  properties: {
+    subnet: {
+      id: resourceId('Microsoft.Network/virtualNetworks/subnets', virtualNetwork.name, privateEndpointSubnetName)
+    }
+    privateLinkServiceConnections: [
+      {
+        name: 'MyStorageTablePrivateLinkConnection'
+        properties: {
+          privateLinkServiceId: storageAccount.id
+          groupIds: [
+            'table'
+          ]
+        }
+      }
+    ]
+  }
+}
+
+resource storageQueuePrivateEndpoint 'Microsoft.Network/privateEndpoints@2021-02-01' = {
+  name: privateEndpointStorageQueueName
+  location: location
+  properties: {
+    subnet: {
+      id: resourceId('Microsoft.Network/virtualNetworks/subnets', virtualNetwork.name, privateEndpointSubnetName)
+    }
+    privateLinkServiceConnections: [
+      {
+        name: 'MyStorageQueuePrivateLinkConnection'
+        properties: {
+          privateLinkServiceId: storageAccount.id
+          groupIds: [
+            'queue'
+          ]
+        }
+      }
+    ]
+  }
+}
+
+resource storageBlobPrivateEndpoint 'Microsoft.Network/privateEndpoints@2021-02-01' = {
+  name: privateEndpointStorageBlobName
+  location: location
+  properties: {
+    subnet: {
+      id: resourceId('Microsoft.Network/virtualNetworks/subnets', virtualNetwork.name, privateEndpointSubnetName)
+    }
+    privateLinkServiceConnections: [
+      {
+        name: 'MyStorageBlobPrivateLinkConnection'
+        properties: {
+          privateLinkServiceId: storageAccount.id
+          groupIds: [
+            'blob'
+          ]
+        }
+      }
+    ]
+  }
+}
+
+resource storageAccount 'Microsoft.Storage/storageAccounts@2021-02-01' = {
+  name: functionStorageAccountName
+  location: location
+  kind: 'StorageV2'
+  sku: {
+    name: 'Standard_LRS'
+  }
+  properties: {
+    networkAcls: {
+      bypass: 'None'
+      defaultAction: 'Deny'
+    }
+  }
+  dependsOn: [
+    virtualNetwork
+  ]
+}
+
+resource functionContentShare 'Microsoft.Storage/storageAccounts/fileServices/shares@2021-04-01' = {
+  name: '${storageAccount.name}/default/${functionContentShareName}'
+  dependsOn: [
+    storageAccount
+  ]
+}
+
+resource appInsights 'Microsoft.Insights/components@2020-02-02-preview' = {
+  name: applicationInsightsName
+  location: location
+  kind: 'web'
+  properties: {
+    Application_Type: 'web'
+  }
+}
+
+resource plan 'Microsoft.Web/serverfarms@2021-01-01' = {
+  location: location
+  name: functionAppPlanName
+  sku: {
+    name: functionAppPlanSku
+    tier: 'ElasticPremium'
+    size: functionAppPlanSku
+    family: 'EP'
+  }
+  kind: 'elastic'
+  properties: {
+    maximumElasticWorkerCount: 20
+    reserved: isReserved
+  }
+}
+
+resource function 'Microsoft.Web/sites@2021-01-01' = {
```

Comment:
function is pretty generic - let's use "functionApp" or something more descriptive

Reply:
Done.